src/Data/List/Relation/Unary/AllPairs/Properties.agda

```diff
@@ -132,3 +135,16 @@ module _ {R : Rel A ℓ} {P : Pred A p} (P? : Decidable P) where
   filter⁺ {x ∷ xs} (x∉xs ∷ xs!) with does (P? x)
   ... | false = filter⁺ xs!
   ... | true  = All.filter⁺ P? x∉xs ∷ filter⁺ xs!
+
+------------------------------------------------------------------------
+-- pointwise
+
+module _ {R : Rel A ℓ}  where
+
+  pointwise⁺ : {xs : List (Maybe A)} → AllPairs (Pointwise R) xs → AllPairs R (catMaybes xs)
+  pointwise⁺ {xs = []} [] = []
+  pointwise⁺ {xs = nothing ∷  _} (x∼xs ∷ pxs) = pointwise⁺ pxs
+  pointwise⁺ {xs = just x  ∷ xs} (x∼xs ∷ pxs) = catMaybes⁺ (All.map helper x∼xs) ∷ pointwise⁺ pxs
+    where
+    helper : Pointwise R (just x) ⊆ Any (R x)
```

Comment:
This seems worth lifting to the toplevel in either `Data.Maybe.Relation.Binary.Pointwise`  or `Data.Maybe.Relation.Unary.Any`